src/Components/Components/src/Reflection/ComponentProperties.cs

```diff
@@ -259,7 +262,9 @@ public WritersForType([DynamicallyAccessedMembers(Component)] Type targetType)
             {
                 var parameterAttribute = propertyInfo.GetCustomAttribute<ParameterAttribute>();
                 var cascadingParameterAttribute = propertyInfo.GetCustomAttribute<CascadingParameterAttribute>();
-                var isParameter = parameterAttribute != null || cascadingParameterAttribute != null;
+                var supplyParameterFromFormAttribute = propertyInfo.GetCustomAttribute<SupplyParameterFromFormAttribute>();
+
+                var isParameter = parameterAttribute != null || cascadingParameterAttribute != null || supplyParameterFromFormAttribute != null;
```

Comment:
Are we going to require that `[SupplyParameterFromForm]` also requires the `[Parameter]` attribute? This would be consistent with `[SupplyParameterFromQuery]`. If so, should we remove the change to `isParameter`?

Reply:
I would hope that `[SupplyParameterFrom***]` does not require anything extra. It already clearly expresses intent, and its "quite a mouthful" to type.

(In fact, we might even abbreviate some of this to `[QueryParameter]`, `[FormParameter]` and so on).

What do you think?

Reply:
Great question! The reasoning for `[SupplyParameterFromQuery]` having to be combined with `[Parameter]` was given in https://github.com/dotnet/aspnetcore/issues/33338 as follows:

> The benefit of it actually being a [Parameter] property (rather than just using [FromQuery] alone) is that the whole parameter-supplying mechanism will just work following its normal semantics about notifying-only-if-changed. All existing mechanisms already know how to deal with this.

I think **that reasoning does not apply if we are changing to a CascadingValue-based solution** since the parameter updating logic will automatically work.

So yes I agree it would be good *not* to require people to also specify `[Parameter]`. However we should definitely verify we can make this work with `[SupplyParameterFromQuery]` too otherwise it's going to be super weird, i.e.:

 * Existing code that continues to have `[Parameter, SupplyParameterFromQuery]` should still work even if `[SupplyParameterFromQuery]` starts being built on a CascadingValue-based solution
 * `[SupplyParameterFromQuery]` should start working too, so people writing new code can just write it more succinctly

AFAIK that should be viable but @javiercn would know more precisely since he's putting together this CascadingValue-based solution and will be able to check it can also extended to work for `[SupplyParameterFromQuery]` in this way.

Reply:
> (In fact, we might even abbreviate some of this to [QueryParameter], [FormParameter] and so on).

I really like those names.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/48443 to track